### DISCUSSION TITLE
Attention mask/bias

### DISCUSSION THREAD

**Discussion by YotamNitzan** [PixArt-alpha/PixArt-alpha#24] [Q&A] [answered]
Thanks for releasing the code and models!
I'm working through your code and trying to understand the role of the attention mask. I see it's loaded by the [dataloader](https://github.com/PixArt-alpha/PixArt-alpha/blob/a4f8fd334ded518103aad9cfb76983560a7bc584/diffusion/data/datasets/InternalData_ms.py#L119) from  the `caption_feature_wmask\` directory. Then it's passed all the way to the cross-attention where it's [used](https://github.com/PixArt-alpha/PixArt-alpha/blob/a4f8fd334ded518103aad9cfb76983560a7bc584/diffusion/model/nets/PixArt_blocks.py#L53) as a bias for the attention mask.

Can you please explain what is the role of this attention mask and how is it created?

**Comment by lawrence-cj** [comment, marked as answer]
The T5 model is designed to generate a feature matrix of size `120x4096`, irrespective of the sentence length. However, the feature matrix might include vectors `devoid of semantic information` that are not entirely zero. Hence, we require a mask to prevent these vectors from influencing the attention calculations during both training and inference processes.
